lib/grpc/transport/http2.ex

```diff
@@ -151,8 +171,26 @@ defmodule GRPC.Transport.HTTP2 do
     {key, URI.decode(val)}
   end
 
+  defp decode_reserved({"grpc-status-details-bin" = key, val}) do
+    {key, decode_status_details(val)}
+  end
+
   defp decode_reserved(kv), do: kv
 
+  @spec decode_status_details(String.t()) :: [Google.Protobuf.Any.t()] | nil
+  def decode_status_details(encoded_details) when is_binary(encoded_details) do
+    try do
+      encoded_details
+      |> Base.decode64!(padding: false)
+      |> Google.Rpc.Status.decode()
```

Comment:
this is not currently available, should we upstream this to https://github.com/elixir-protobuf/protobuf or include it in here?

Reply:
Checked other languages and it's very inconsistent:

1. Some languages (Go, C#, Java) vendor all Google API protobuf messages as one package.
2. Some languages (Python) vendor the Status message separately.
3. Some languages (Ruby, JS, Rust) require you to generate the message yourself.

We could:

1. Vendor it into this library.
2. Create a separate library with Google API protobufs and a) make it a strict dependency of this library, b) make it an optional dependency of this library (if it's loaded then `grpc-status-details-bin` gets decoded).